src/lib/dnssd/Resolver_ImplMinimalMdns.cpp

```diff
@@ -372,7 +370,23 @@ void MinMdnsResolver::AdvancePendingResolverStates()
 
         if (missing.Has(IncrementalResolver::RequiredInformationBitFlags::kIpAddress))
         {
-            ScheduleIpAddressResolve(resolver->GetTargetHostName());
+            if (resolver->IsActiveBrowseParse())
+            {
+                // Browse wants IP addresses
+                ScheduleIpAddressResolve(resolver->GetTargetHostName());
+            }
+            else if (mActiveResolves.ShouldResolveIpAddress(resolver->OperationalParsePeerId()))
```

Comment:
How long does an active resolve last?

Reply:
There are timeouts, from what I can see ActiveResolveAttempts has a kMaxRetryDelay of 16 and we use exponential backoff. we start at 1, so we  will do 1,2,4,8,16 and then give up. So 31 seconds timeout.